### PR TITLE
do hw

### PR DESCRIPTION
**Pull request by RetKinf**

### CODE REVIEW

File: src/main/java/com/example/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private String description;
+    private String coverImage;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
```
Comment (okuzan) [inline src/main/java/com/example/model/Book.java:20]: use lombok's @ Getter and @ Setter

File: src/main/java/com/example/service/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    private BookRepository bookRepository;
+
+    @Override
+    public Book save(Book book) {
+        return null;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return List.of();
```
Comment (okuzan) [inline src/main/java/com/example/service/BookServiceImpl.java:17]: add implementations

File: src/main/java/com/example/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import java.math.BigDecimal;
+
+public class Book {
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (okuzan) [inline src/main/java/com/example/model/Book.java:12]: image
enforce nullability, use jakarta.persistence.Id; etc

File: src/main/java/com/example/model/Book.java (hunk 1 above)
Comment (RetKinf) [reply src/main/java/com/example/model/Book.java:20]: Lombok has not been studied yet

File: src/main/java/com/example/repository/BookRepository.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.example.repository;
+
+import com.example.model.Book;
+import java.util.Optional;
+
+public interface BookRepository {
+    Book save(Book book);
+
+    Optional<Book> findAll();
```
Comment (kerrrusha) [inline src/main/java/com/example/repository/BookRepository.java:9]: invalid method return type
```suggestion
    List<Book> findAll();
```

File: src/main/java/com/example/service/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    @Override
+    public Book save(Book book) {
+        return bookRepository.save(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return List.of(
+                bookRepository.findAll()
+                        .orElseThrow(() -> new EntityNotFoundException("Book not found")
```
Comment (kerrrusha) [inline src/main/java/com/example/service/BookServiceImpl.java:20]:
```suggestion
```

**Review by kerrrusha — CHANGES_REQUESTED** [review]

**Review by kerrrusha — APPROVED** [review]

**Pull request merged**